src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -395,6 +397,35 @@ fn make_core(
   )
 }
 
+fn workunits_to_py_tuple_value(workunits: &HashSet<WorkUnit>) -> Value {
```

Comment:
Extremely excited to see this method here! As a note, I would (maybe) consider in future work an approach somewhat like the one described in #7371 if we end up finding that we later need to change the definition of this method too often in response to changes in python code.